src/client/shader.cpp

```diff
@@ -790,7 +790,8 @@ ShaderInfo ShaderSource::generateShader(const std::string &name,
 		dumpShaderProgram(warningstream, "Vertex", vertex_shader);
 		dumpShaderProgram(warningstream, "Fragment", fragment_shader);
 		dumpShaderProgram(warningstream, "Geometry", geometry_shader);
-		return shaderinfo;
+		throw ShaderException("Failed to compile the \"" + name + "\" shader. "
+			"Check the log for details.");
```

Comment:
```suggestion
			"Check debug.txt for details.");
```

for consistency with other error messages